pkg/controller/node/status.go

```diff
@@ -60,24 +89,93 @@ func calculateStatus(cconfig *v1.ControllerConfig, pool *mcfgv1.MachineConfigPoo
 	}
 	machineCount := int32(len(nodes))
 
-	updatedMachines := getUpdatedMachines(pool, nodes)
+	var degradedMachines, readyMachines, updatedMachines, unavailableMachines, updatingMachines []*corev1.Node
+	// if we represent updating properly here, we will also represent updating properly in the CO
+	// so this solves the cordoning RFE and the upgradeable RFE
+	// updating == updatePrepared, updateExecuted, updatedComplete, postAction, cordoning, draining
+	// updated == nodeResumed, updated
+	// ready == nodeResumed, updated
+	// unavailable == draining, cordoned
+	// degraded == if the condition.Reason == error
+	// this ensures that a MCP only enters Upgradeable==False if the node actually needs to upgrade to the new MC
+	for _, state := range mcs {
+		var ourNode *corev1.Node
+		for _, n := range nodes {
+			if state.Name == n.Name {
+				ourNode = n
+				break
+			}
+		}
+		if ourNode == nil {
+			klog.Errorf("Could not find specified node %s", state.Name)
+		}
+		if len(state.Status.Conditions) == 0 {
+			// not ready yet
+			break
+		}
+		for _, cond := range state.Status.Conditions {
+			if strings.Contains(cond.Message, "Error:") {
+				degradedMachines = append(degradedMachines, ourNode)
+				continue
+			}
+			if cond.Status == metav1.ConditionUnknown {
+				switch v1alpha1.StateProgress(cond.Type) {
+				case mcfgalphav1.MachineConfigNodeUpdatePrepared:
```

Comment:
issue: Looking at the code coverage analysis in CI, it looks like these lines are not being exercised by the test suite. Can you add a few test cases to ensure that they are being exercised?

Reply:
Got it, thanks for clarifying!